test-framework/core/src/main/java/org/keycloak/testframework/admin/KeycloakAdminClientFactory.java

```diff
@@ -0,0 +1,97 @@
+package org.keycloak.testframework.admin;
+
+import org.keycloak.OAuth2Constants;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.KeycloakBuilder;
+import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.keycloak.testframework.TestFrameworkException;
+import org.keycloak.testframework.annotations.InjectAdminClientFactory;
+import org.keycloak.testframework.config.Config;
+import org.keycloak.testframework.injection.InstanceContext;
+import org.keycloak.testframework.realm.ManagedRealm;
+import org.keycloak.testframework.realm.ManagedUser;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class KeycloakAdminClientFactory {
+
+    private final List<Keycloak> adminClients = new ArrayList<>();
+
+    private final InstanceContext<KeycloakAdminClientFactory, InjectAdminClientFactory> instanceContext;
+    private final String serverUrl;
+    private final String grantType;
+
+    public KeycloakAdminClientFactory(InstanceContext<KeycloakAdminClientFactory, InjectAdminClientFactory> instanceContext, String serverUrl, String grantType) {
+        this.instanceContext = instanceContext;
+        this.serverUrl = serverUrl;
+        this.grantType = grantType;
+    }
+
+    public Keycloak createMaster() {
+        KeycloakBuilder clientBuilder = KeycloakBuilder.builder()
+                .serverUrl(serverUrl)
+                .grantType(grantType);
+
+        clientBuilder.realm("master").clientId(Config.getAdminClientId()).clientSecret(Config.getAdminClientSecret());
+        Keycloak adminClient = clientBuilder.build();
+        adminClients.add(adminClient);
+        return adminClient;
+    }
+
+    public Keycloak create(String realmRef, String clientId, String user) {
```

Comment:
This won't work as you won't be able to get user password from the Admin APIs. I would just pass the username and password instead.

Reply:
Does it behave any differently than in the `KeycloakAdminClientSupplier`?
https://github.com/keycloak/keycloak/blob/main/test-framework/core/src/main/java/org/keycloak/testframework/admin/KeycloakAdminClientSupplier.java#L63-L69

Reply:
For injected users, we won't be able to get the password, as it uses the same mechanism